### PR TITLE
Preparatory work for certificate generation

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: src/certificate.rs [hunk 1]
```diff
@@ -166,11 +198,11 @@ impl Certificate {
     #[cfg(feature = "untested")]
     pub fn delete(yubikey: &mut YubiKey, slot: SlotId) -> Result<(), Error> {
         let txn = yubikey.begin_transaction()?;
-        write_certificate(&txn, slot, None, 0)
+        write_certificate(&txn, slot, None, CertInfo::Uncompressed)
     }
 
     /// Initialize a local certificate struct from the given bytebuffer
-    pub fn new(cert: impl Into<Buffer>) -> Result<Self, Error> {
+    pub fn from_bytes(cert: impl Into<Buffer>) -> Result<Self, Error> {
```
Comment (tarcieri) [inline src/certificate.rs]: This is cool, but perhaps also add a `impl<'a> TryFrom<&'a [u8]> for Certificate`?

File: src/certificate.rs [hunk 2]
```diff
@@ -57,17 +58,43 @@ const OID_EC_PUBLIC_KEY: &str = "1.2.840.10045.2.1";
...
-
 const TAG_CERT: u8 = 0x70;
 #[cfg(feature = "untested")]
 const TAG_CERT_COMPRESS: u8 = 0x71;
 #[cfg(feature = "untested")]
 const TAG_CERT_LRC: u8 = 0xFE;
 
+/// Information about how a [`Certificate`] is stored within a YubiKey.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum CertInfo {
```
Comment (tarcieri) [inline src/certificate.rs]: Nice

File: src/key.rs [hunk 3]
```diff
@@ -475,7 +443,7 @@ pub fn generate(
     algorithm: AlgorithmId,
     pin_policy: PinPolicy,
     touch_policy: TouchPolicy,
-) -> Result<GeneratedKey, Error> {
+) -> Result<PublicKeyInfo, Error> {
```
Comment (tarcieri) [inline src/key.rs]: 👍

File: src/certificate.rs (hunk 1 above)
Comment (str4d) [reply src/certificate.rs]: Done.

**Pull request merged**